packages/components/addon/components/hds/dialog/index.hbs

```diff
@@ -0,0 +1,12 @@
+{{! template-lint-disable no-invalid-interactive }}
+<dialog
```

Comment:
I _think_ we should break down the component in three parts:

- the `<dialog> element itself (and children)
- the backdrop element
- the union of the two in the full modal component

(class names to be discussed/decided)

More than happy to brainstorm on this.

Reply:
I know we touched on this offline but leaving a comment for posterity.

Indeed Safari 14 and below doesn't offer support for the `<dialog>` element, so I added a polyfill (developed by a few folks at Google a few years back), for now, so we can make the most of this relatively new and admirable alignment in standard across current versions of the major browsers, with the hope to remove the polyfill in 1 year time or so.

However, as you will see in the code, I decided not to use the standard `::backdrop` pseudoelement, but a classic 'overlay' element. I did this so I can remove the additional logic around modal dismissal when 'clicking outside' – as you suggested in the other comment – and leveraged the `focus-trap` library to handle the 'clicking outside' events.